aicsimageio/readers/lif_reader.py

```diff
@@ -0,0 +1,751 @@
+
+    @staticmethod
+    def get_pixel_type(meta: _Element, scene: int = 0) -> np.dtype:
+        """
+        This function parses the metadata to assign the appropriate numpy.dtype
+
+        Parameters
+        ----------
+        meta: lxml.etree.Element
+            The root Element of the metadata etree
+        scene: int
+            The index of the scene, scenes could have different storage data types.
+
+        Returns
+        -------
+        numpy.dtype
+            The appropriate data type to construct the matrix with.
+
+        """
+        ############################
+        #
+        #  This is super weird to me that resolution=12 and resolution=16 would have different
+        #  byte order meaning given 2 uint8s/chars => AB from the byte array one method combines them
+        #  with ( A << 8 | B ) and the other combines them with ( B << 8 | A ).
+        #  This could be wrong but I haven't found any documentation on the image byte order etc so it's
+        #  currently my best guess.
+        #
+        ############################
+        p_types = {8: np.uint8,
+                   12: np.dtype('>u2'),  # big endian uint16
+                   16: np.dtype('<u2'),  # little endian uint16
+                   32: np.dtype('<u4'),  # little endian uint32 ** untested
+                   64: np.dtype('<u8')   # little endian uint64 ** untested
+                   }
+        img_sets = meta.findall(".//Image")
+
+        img = img_sets[scene]
+        chs = img.findall(".//ChannelDescription")
+        resolution = set([ch.attrib['Resolution'] for ch in chs])
+        if len(resolution) != 1:
+            raise exceptions.InconsistentPixelType(f"Metadata contains two conflicting "
+                                                   f"Resolution attributes: {resolution}")
+
+        return p_types[int(resolution.pop())]
+
+    def get_channel_names(self, scene: int = 0) -> List[str]:
+        """
+        Get the channel names for the scene
+
+        Parameters
+        ----------
+        scene: int
+            The index of the scene from which to retrieve the channel names
+
+        Returns
+        -------
+        List[str]
+            A list of descriptive names of the channels of the form "Gray--TL-BF--EMP_BF" and "Green--FLUO--GFP"
+        """
+        # for the remove to work the
+        img_sets = self.metadata.findall(".//Image")
+
+        img = img_sets[scene]
+        scene_channel_list = []
+        chs = img.findall(".//ChannelDescription")
+        chs_details = img.findall(".//WideFieldChannelInfo")
+        for ch in chs:
+            ch_detail = next(x for x in chs_details if x.attrib["LUT"] == ch.attrib["LUTName"])
+            scene_channel_list.append((f"{ch_detail.attrib['LUT']}--{ch_detail.attrib['ContrastingMethodName']}"
+                                       f"--{ch_detail.attrib['FluoCubeName']}"))
+        return scene_channel_list
```

Comment:
_If_ the channel dimension is always present in LIF files I have no issue with this. _If not_ then this should be `Optional[List[str]]` where if channel dimension doesn't exist in the file it returns `None`.

Reply:
I see you are always removing the channel dimension so this is fine.